src/controllers/presentationController.ts

```diff
@@ -61,12 +62,22 @@ const retrievePresentationStatus = (req: Request, res: Response, next: NextFunct
       })
 };
 
-const updatePresentationStatus = (req: Request, res: Response) => {
-  //status response
-  res.status(201).json({ message: 'method not implemented yet' }); // presentation status
+const updatePresentationStatus = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    validateProperties(requestedStatusProperties, req);
+    if (req.params['id'] !== req.body.presentation_id) {
+      throw new ApiError('presentation_id must be the same in request path parameter and body');
+    }
+    return getMongoRepository(StatusWrapperEntity).updateOne({ presentation_id: req.params['id'] }, { $set: req.body }, { upsert: true })
+        .then(data => {
+            res.redirect(`${req.protocol}://${req.headers.host}${req.baseUrl}${req.path}`)
+        })
+  } catch(error) {
+    next(error)
+  }
 };
 
 PRESENTATION_CONTROLLER.post('/presentations', createPresentation);
 PRESENTATION_CONTROLLER.get('/presentations/:id', retrievePresentation);
-PRESENTATION_CONTROLLER.post('/presentations/:id/statuses', updatePresentationStatus);
+PRESENTATION_CONTROLLER.put('/presentations/:id/statuses', updatePresentationStatus);
```

Comment:
If I am not mistaken as it was many, many months ago, there was a deliberate choice to deviate from normal REST behavior a bit, by using a POST. Think it had something to do with callbacks.

Reply:
reverted to POST.